speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,483 @@
+    save_path : str
+        save directory of the downloaded model.
+    cache_dir: str
+        cache directory used by HuggingFace transformers.
+    freeze : bool (default: True)
+        If `True`, the model is frozen. If `False`, the model will be trained
+        alongside with the rest of the pipeline.
+    freeze_encoder : bool (default: False)
+        If `True`, the model encoder is frozen. If the model is set to encoder only,
+        using the freeze parameter will also freeze the encoder.
+    freeze_feature_extractor: bool (default: False)
+        Whether or not to freeze the feature extractor of SpeechT5.
+    sampling_rate: int (default: 16000)
+        Sampling rate of the audio inputs.
+    encoder_only : bool (default: False)
+        If `True`, will only load the (speech) encoder of the SpeechT5 model.
+    output_attentions: bool (default: True)
+        Whether to output the attentions from the encoder.
+    output_all_hiddens: bool (default: True)
+        Whether or not to output all the hidden states from the encoder. They will
+        be stacked if returned.
+    device : any, optional
+        Device to migrate the model to.
+    *args : tuple
+        Variable-length positional arguments. Each argument can be of any type.
+    **kwargs : dict
+        Variable-length keyword arguments. Each argument name is a string, and
+        each value can be of any type.
+
+    Example
+    -------
+    >>> model_hub = "microsoft/speecht5_asr"
+    >>> save_path = "tmp"
+    >>> model = SpeechT5ForASR(model_hub, save_path=save_path) # doctest: +SKIP
+    """
+
+    def __init__(
+        self,
+        source: str,
+        save_path: str = "",
+        cache_dir: str = "",
+        freeze: bool = False,
+        freeze_encoder: bool = False,
+        freeze_feature_extractor: bool = False,
+        sampling_rate: int = 16_000,
+        encoder_only: bool = False,
+        output_attentions: bool = False,
+        output_all_hiddens: bool = False,
+        device=None,
+        *args,
+        **kwargs,
+    ) -> None:
+
+        if encoder_only and freeze:
+            freeze_encoder = True
+
+        if encoder_only and freeze_encoder:
+            freeze = True
+
+        super().__init__(
+            source=source,
+            save_path=save_path,
+            freeze=freeze,
+            cache_dir=cache_dir,
+            device=device,
+            *args,
+            **kwargs,
+        )
+
+        self.sampling_rate = sampling_rate
+        self.encoder_only = encoder_only
+        self.freeze_encoder = freeze_encoder
+        self.freeze_feature_extractor = freeze_feature_extractor
+        self.output_attentions = output_attentions
+        self.output_all_hiddens = output_all_hiddens
+        self.generation_config = None
+
+        self.load_tokenizer(source=source, **kwargs)
+
+        config = SpeechT5Config.from_pretrained(
+            source, cache_dir=cache_dir, **kwargs
+        )
```

Comment:
not sure to understand why do you have to fetch the config? Isnt it already fetch here: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/lobes/models/huggingface_transformers/huggingface.py#L113-L117 ?